game-core/src/main/java/games/strategy/triplea/delegate/battle/casualty/CasualtySelector.java

```diff
@@ -74,62 +73,22 @@ public static CasualtyDetails selectCasualties(
     if (targetsToPickFrom.isEmpty()) {
       return new CasualtyDetails();
     }
-    if (!friendlyUnits.containsAll(targetsToPickFrom)) {
-      throw new IllegalStateException(
-          "friendlyUnits should but does not contain all units from targetsToPickFrom"
-              + ", battlesite: "
-              + battlesite
-              + ", friendlyUnits: "
-              + friendlyUnits
-              + ", targetsToPickFrom: "
-              + targetsToPickFrom);
-    }
     final GameData data = bridge.getData();
-    final boolean isEditMode = BaseEditDelegate.getEditMode(data);
     final Player tripleaPlayer =
         player.isNull() ? new WeakAi(player.getName()) : bridge.getRemotePlayer(player);
     final Map<Unit, Collection<Unit>> dependents =
         headLess ? Map.of() : CasualtyUtil.getDependents(targetsToPickFrom);
-    if (isEditMode && !headLess) {
-      final CasualtyDetails editSelection =
-          tripleaPlayer.selectCasualties(
-              targetsToPickFrom,
-              dependents,
-              0,
```

Comment:
This zero is a hard-coding of the number of hits (which creates problems).
We could in theory have kept this behavior and simplified to set CasualtyDetails to be empty as the 'editSelection' result is always (and forced to be empty).

Perhaps the original intent of this code was to prompt zero casualties and allow for selection of any number? Regardless if that was the intent or not,  you are forced to choose zero casualties which creates dead code paths in casualty selection (always returns empty).

Reply:
So if you want to simplify this code I'm good with that but I think we should keep the edit mode behavior.

Reply:
I really disagree, something that can cause a game-crash without possibility to save is very bad. How do we resolve this disagreement? 

The updates here are blocking further updates from being done.